internal/targetedretries/dot_net_xunit_substitution.go

```diff
@@ -67,7 +67,9 @@ func (s DotNetxUnitSubstitution) SubstitutionsFor(
 	}
 
 	if len(tests) > 0 {
-		return []map[string]string{{"filter": strings.Join(tests, " | ")}}, nil
+		return []map[string]string{{
+			"filter": strings.ReplaceAll(strings.Join(tests, " | "), "\"", "\\\""),
```

Comment:
I think it might be more inline with how we handle escaping in the rest of the codebase if we instead used `templating.ShellEscape` on the formatted test (line 60) like we do in the other substitutions: https://github.com/rwx-research/captain/blob/6aea85538ccb37a13f1f6f332fc35e76fb31a672/internal/targetedretries/javascript_jest_substitution.go#L73

Specifically, that function shell-escapes single quotes and our expected usage looks more like [`dotnet test --filter '{{ filter }}'`](https://www.rwx.com/docs/captain/test-frameworks/dot-net/xunit#retrying-tests) (we tended to lean towards expecting single quotes to avoid incidental bash interpolation)

Reply:
Done, I think this PR should remain open until it can be verified against a fixed `dotnet test`.